CMakeLists.txt

```diff
@@ -546,5 +546,7 @@ if(NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
 endif()
 
 add_custom_target(format
-    COMMAND clang-format -i `git ls-files *.hpp *.inl *.cpp`
+    COMMAND "git ls-files -- '*.hpp' '*.inl' '*.cpp' ':!:extlibs/*' -z | xargs -0 clang-format -i"
+    # -z makes git separate lines with null, instead of \n
+    # -0 makes xargs separate by null, not \n, and makes other characters be taken literally
```

Comment:
```
thrasher ~/Projects/sfml$ git ls-files -- '*.hpp' '*.inl' '*.cpp' ':!:extlibs/*' -z | xargs -0 clang-format -i
File name too long
```
Something is going wrong for me when I run this.

Reply:
oh... lemme check that

Reply:
Do we need quotes around this command or not? I don't think we do.

Reply:
`-z` needs to be before `--` to be interpreted as an option.

Reply:
> -z needs to be before -- to be interpreted as an option.

This fixed my problem and now clang-format runs.

Reply:
> `-z` needs to be before `--` to be interpreted as an option.

ahh yes, nice spot



> Do we need quotes around this command or not? I don't think we do.

im not exactly sure how cmake might handle the different `'`s inside it, I added them just to be safe

Reply:
It actually won't for me until I remove the quotes so safe to say we don't need them.

Reply:
Ahh my bad, removed.